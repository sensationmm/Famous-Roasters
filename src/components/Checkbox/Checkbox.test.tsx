import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'

import { Checkbox } from '.'

global.alert = jest.fn()

describe('Checkbox component', () => {
  it('Renders correctly', async () => {
    const { container } = render(<Checkbox name="test" text="Dieser Kaffee ist lekka" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for selected', async () => {
    const { container } = render(<Checkbox name="test" text="Dieser Kaffee ist lekka" selected={true} />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can toggle the state by clicking', async () => {
    render(<Checkbox name="test" text="Dieser Kaffee ist lekka" toggleSelected={() => alert('works')} />)
    const checkbox = await screen.findByTestId('checkbox')
    expect(checkbox).toBeInTheDocument()
    fireEvent.click(checkbox)
    fireEvent.click(checkbox)
  })
})
