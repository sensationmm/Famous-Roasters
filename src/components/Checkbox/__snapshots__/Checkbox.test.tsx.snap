// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Renders correctly 1`] = `
<div>
  <button
    class="flex w-full flex-row items-start cursor-pointer disabled:opacity-20 disabled:cursor-not-allowed"
    data-testid="checkbox"
  >
    <div>
      <svg
        class="undefined"
        fill="inherit"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M19 22H5c-1.7 0-3-1.3-3-3V5c0-1.7 1.3-3 3-3h14c1.7 0 3 1.3 3 3v14c0 1.7-1.3 3-3 3zM5 4c-.6 0-1 .4-1 1v14c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1H5z"
          fill="inherit"
          fill-rule="evenodd"
          stroke="inherit"
          stroke-width="0.25"
        />
      </svg>
    </div>
    <div
      class="flex grow text-left pl-2 text-base leading-6"
    >
      Dieser Kaffee ist lekka
    </div>
  </button>
</div>
`;

exports[`Checkbox component Renders correctly for selected 1`] = `
<div>
  <button
    class="flex w-full flex-row items-start cursor-pointer disabled:opacity-20 disabled:cursor-not-allowed"
    data-testid="checkbox"
  >
    <div>
      <svg
        class="undefined"
        fill="inherit"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M19 2H5C3.3 2 2 3.3 2 5v14c0 1.7 1.3 3 3 3h14c1.7 0 3-1.3 3-3V5c0-1.7-1.3-3-3-3z M9.4 17.6c-.3 0-.5-.1-.7-.3l-3.4-3.4c-.4-.4-.4-1 0-1.4s1-.4 1.4 0l2.7 2.7 7.9-7.9c.4-.4 1-.4 1.4 0s.4 1 0 1.4l-8.6 8.6c-.2.2-.4.3-.7.3z"
          fill="inherit"
          fill-rule="evenodd"
          stroke="inherit"
          stroke-width="0.25"
        />
      </svg>
    </div>
    <div
      class="flex grow text-left pl-2 text-base leading-6"
    >
      Dieser Kaffee ist lekka
    </div>
  </button>
</div>
`;
